include/deal.II/base/tensor_base.h

```diff
@@ -116,14 +115,28 @@ class Tensor<0,dim,Number>
   typedef typename numbers::NumberTraits<Number>::real_type real_type;
 
   /**
-   * Constructor. Set to zero.
+   * The tensor type this object represents. In the special case of a
+   * tensor or rank 0 we strip the tensor class and just store the scalar
+   * object.
    */
-  Tensor ();
+  typedef Number tensor_type;
```

Comment:
This strikes me as a bad idea. The scalar itself is not a good representation of a tensor. I would suggest
```
  typedef Tensor<0,dim,Number> tensor_type;
```